### ISSUE TITLE
Reference script limitations are incomplete

### ISSUE THREAD

**Issue by WhatisRT** [IntersectMBO/formal-ledger-specifications#537]
We need to add an upper bound on the reference scripts allowed per transaction and per block. Additionally, the cost was changed in the implementation to a non-linear formula (see `getConwayMinFeeTx` in https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs).

**Comment by jmchapman** [comment]
Do we have a size estimate for this?

**Comment by WhatisRT** [comment]
This is pretty small, should be doable in 1-2 days.

**Comment by williamdemeo** [comment]
@jmchapman @WhatisRT This would be much smaller/easier task if the description were less vague.  We discussed the issue briefly in the last ledger meeting, but we didn't specify, for example, what should the upper bound be?

Also, it would be very helpful to me if more things about the issue were made explicit, like which files (functions, types, etc) will likely need modifying.

Re. "an upper bound on the reference scripts allowed per transaction and per block," does this mean an upper bound on the *number* of reference scripts, or the size of a reference script?  (This may end up being a stupid question, but based on past experience, it's important that I clear up any possible misunderstandings in advance.)

To which module will this bound likely be added?

Re. "the cost was changed in the implementation to a non-linear formula (see getConwayMinFeeTx in https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs)" 
I believe in the meeting you indicated that we need to implement the new cost formula and probably add the formula to the appendix of the pdf.  Where is the current cost function?  Where should this change go?

After some digging around I could probably guess, with a relatively high degree of accuracy, the answers to some of these questions, but I suspect the answers are obvious to you and by including them in the issue description you would save me a lot of time and ensure that I can get the task done withing the time frame you suggest above.
